cachi2/core/package_managers/rubygems/parser.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from typing import Optional, Union
+
+import pydantic
+
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(__name__)
+
+GEMFILE_LOCK = "Gemfile.lock"
+
+
+class GemMetadata(pydantic.BaseModel, extra="forbid"):
```

Comment:
I thought we could distinguish between supported classes for better readability and scalability.
Compare to one `dataclass` in [cachito](https://github.com/containerbuildsystem/cachito/blob/master/cachito/workers/pkg_managers/rubygems.py#L42).

Reply:
I like having these classes separate, but I think Taylor's point is that they could simply extend `NamedTuple` or be a `dataclass`, instead of relying on Pydantic.

Reply:
Okay, I can change it...I assume we won't add any validators or any extra features to those classes, right ?

Reply:
I don't think we'll need validators. Their content will always come from the gemlock-parser, so it is pretty much predictable data at this point.

Reply:
Are you okay with `dataclass` ? `NamedTuple` does not feel to me like something that should be extendable or have a base class.